src/terminal.rs

```diff
@@ -25,53 +34,153 @@ impl Display for TerminalError {
+                    return Ok(Action::Update(id, text));
+                }
+                'd' => return Ok(Action::Delete(id)),
+                'x' => return Ok(Action::Exit),
+                _ => println!("{}_>> Desculpa eu não entendi.", Emoji("🤨", ":/")),
+            }
+        }
+    }
+
+    pub fn ask_for_action(&self) -> Result<Action, TerminalError> {
+        println!("\nAguarde ...");
+        thread::sleep(Duration::from_millis(2000));
+        self.term.clear_screen().map_err(TerminalError::Stdout)?;
         println!(
-            "{}_>> Olá, gostaria de adicionar um novo TODO? (s/n)",
-            Emoji("😃", ":)")
+            "################# {} ################# \n\n",
+            "BEM VINDO AO TODO CLI"
+        );
+        println!("{}_>> Olá, como posso te ajudar?", Emoji("😃", ":)"));
+        println!(
+            "{} >> Digite 'a' para adicionar um novo TODO",
+            Emoji("✅", ":)")
         );
+        println!("{} >> Digite 'l' para listar os TODOs", Emoji("📝", ":)"));
+        println!("{} >> Digite 'x' para sair", Emoji("👋", ":)"));
         loop {
             let answer = self.term.read_char().map_err(TerminalError::Stdin)?;
             match answer {
-                's' => return Ok(Some(self.add_todo()?)),
-                'n' => return Ok(None),
-                'x' => return Err(TerminalError::Test("AlphaEdtech & TerraMagna".to_string())),
-                _ =>  println!("{}_>> Desculpa eu não entendi. Digite 's' se deseja adicionar um novo TODO ou 'n' se deseja sair. ", Emoji("🤨",":/"))
+                'a' => return Ok(Action::Add),
+                'l' => return Ok(Action::List),
+                'x' => return Ok(Action::Exit),
+                'w' => return Err(TerminalError::Test("AlphaEdtech & TerraMagna".to_string())),
+                _ =>  println!("{}_>> Desculpa eu não entendi. Digite 'a' para adicionar um novo TODO, 'l' para listar os TODOs ou 'x' para sair. ", Emoji("🤨",":/"))
             }
         }
     }
 
-    fn add_todo(&mut self) -> Result<Todo, TerminalError> {
-        println!("{} >> Qual é o TODO?", Emoji("😃", ":)"));
+    pub fn add_todo(&self) -> Result<String, TerminalError> {
+        self.term.clear_screen().map_err(TerminalError::Stdout)?;
+        println!(
+            "################# {} ################# \n\n",
+            "ADICIONAR TODO"
+        );
+
+        println!(
+            "{} >> Qual é o novo TODO que gostaria de adicionar?",
+            Emoji("😃", ":)")
+        );
         let message = self.input()?;
-        Ok(Todo::new(message))
+        Ok(message)
     }
 
-    fn show_todo(&mut self, todo: &Todo) -> Result<(), TerminalError> {
+    pub fn list_todos(&self, list: Vec<&Todo>) -> Result<Option<String>, TerminalError> {
+        self.term.clear_screen().map_err(TerminalError::Stdout)?;
+        println!(
+            "################# {} ################# \n\n",
+            "LISTAGEM DOS TODOS"
+        );
```

Comment:
Dado que não há nenhuma estilização ou coisa do genero em "LISTAGEM ...", podemos remover a formatação.

```suggestion
        println!("################# LISTAGEM DOS TODOS ################# \n\n");
```

Reply:
Na verdade, era para ter criado um método que recebia o valor do titulo e estilizar.... Mas a execução passou longe da ideia. Acabei dando só copia e colar . Vou ajeitar também.